test/coinselector-test.js

```diff
@@ -0,0 +1,166 @@
+      116000, 110000, 109000, 108000, 106000, 105000, 101000, 98000, 96000, 90000,
+      85000, 82000, 81000, 80000, 78000, 71000, 67000, 66000, 63000, 55000, 53000,
+      51000, 45000, 44000, 41000, 38000, 36000, 23000, 19000, 16000, 11000, 6000];
+    for (const target of targets) {
+      it(`should select target=${target} using Branch and Bound`, () => {
+        const selection = selector.selectBnB(target, costOfChange);
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += utxos[i].effectiveValue;
+        }
+        assert.strictEqual(selectedValues, target);
+      });
+    }
+
+    // these should fail because we are using (target - 500)
+    for (const target of targets) {
+      it(`should fail to select target=${target - 500} using Branch and Bound`, () => {
+        const selection = selector.selectBnB(target - 500, costOfChange);
+        assert.strictEqual(selection.length, 0);
+      });
+    }
+
+    // these targets have multiple solutions
+    targets = [150000, 130000, 101000, 50000, 15000, 13000, 5000, 3000];
+    for (const target of targets) {
+      it(`should select more inputs in low feerate environment, target=${target}`, () => {
+        selector.rate = 5000;
+        const lowFeeSelection = selector.selectBnB(target, costOfChange);
+        let selectedValues = 0;
+        for (const i of lowFeeSelection) {
+          selectedValues += utxos[i].effectiveValue;
+        }
+        assert.strictEqual(selectedValues, target);
+        selector.rate = 6000;
+        const highFeeSelection = selector.selectBnB(target, costOfChange);
+        selectedValues = 0;
+        for (const i of highFeeSelection) {
+          selectedValues += utxos[i].effectiveValue;
+        }
+        assert.strictEqual(selectedValues, target);
+        assert(lowFeeSelection.length > highFeeSelection.length);
+      });
+    }
+  });
+
+  describe('Lowest Larger Selection', function() {
+    // try selecting a single UTXO
+    for (const value of values) {
+      it(`should select target=${value} using Lowest Larger`, () => {
+        const selection = selector.selectLowestLarger(value);
+        assert.strictEqual(selection.length, 1);
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += utxos[i].effectiveValue;
+        }
+        assert.strictEqual(selectedValues, value);
+      });
+    }
+
+    // these targets may or may not have exact solutions
+    targets = [221000, 220000, 215000, 214000, 211000, 208000, 206000, 203000, 201000,
+      195000, 186000, 178000, 166000, 160000, 155000, 152000, 146000, 139000, 119000,
+      116000, 110000, 109000, 108000, 106000, 105000, 101000, 98000, 96000, 90000,
+      85000, 82000, 81000, 80000, 78000, 71000, 67000, 66000, 63000, 55000, 53000,
+      51000, 45000, 44000, 41000, 38000, 36000, 23000, 19000, 16000, 11000, 6000];
```

Comment:
if these are the same targets as BnB test, maybe move them up to top of file as constant? Same goes for other group of targets